DistroLauncher/algorithms.h

```diff
@@ -130,3 +130,36 @@ template <typename T, typename... Args> void push_back_many(std::vector<T>& vec,
     vec.reserve(vec.size() + sizeof...(args));
     (vec.push_back(std::forward<Args>(args)), ...);
 }
+
+// Follows the semantics of std::getline(stream&, string&) while implemented for an iterator instead of the stream
+// itself, i.e. reads from [input] iterator until [delim] is found. Although it doesn't go to the string, it's
+// extracted from the underlying stream pointed by the [input] iterator. [str] is also erased before insertion.
+// Iterator is taken by copy, as all algorithms accepting iterators are, but one must remember that the underlying
+// stream linked to the iterator will still have it's internal state changed as the iterator advances, i.e.
+// contents are going to be extracted.
+template <class CharT, class Traits, class Allocator>
+std::istreambuf_iterator<CharT, Traits> getline(std::istreambuf_iterator<CharT, Traits> input,
+                                                std::basic_string<CharT, Traits, Allocator>& str, CharT delim = '\n')
+{
+    str.erase();
+    // missing std::copy_until - a copy_if that stops when the predicate evaluates to true :(
+    while (input != std::istreambuf_iterator<CharT, Traits>{}) {
+        char elem = *input++;
+        if (elem == delim) {
+            break;
+        }
+        str.append(1, elem);
+    }
+    return input;
+    // We could, but let's not make it a template for now (at least until we find a need for reading another char type).
+    // We may find ourselves creating a std::istream just to call widen('\n') on it. Less then optimal.
+}
+
+// Returns a string view that starts on the first non-space element of [str] until its end.
+template <class CharT, class Traits>
+std::basic_string_view<CharT, Traits> left_trimmed(std::basic_string_view<CharT, Traits> str)
+{
+    auto nonSpaceStart = std::find_if_not(str.begin(), str.end(), [](auto elem) { return std::isspace(elem); });
```

Comment:
```suggestion
    auto nonSpaceStart = std::find_if_not(str.begin(), str.end(), std::isspace);
```
Pedantic stuff.

Reply:
That doesn't work because of the const locale parameter :)

```cpp
template <class _Elem>
bool(isspace)(_Elem _Ch, const locale& _Loc) {
    return _STD use_facet<ctype<_Elem>>(_Loc).is(ctype_base::space, _Ch);
}
```

Reply:
So close to being great 😞